src/core/lib/channel/channel_stack.h

```diff
@@ -109,6 +112,19 @@ struct grpc_channel_filter {
      See grpc_call_next_op on how to call the next element in the stack */
   void (*start_transport_stream_op_batch)(grpc_call_element* elem,
                                           grpc_transport_stream_op_batch* op);
+  /* Create a promise to execute one call.
+     If this is non-null, it may be used in preference to
+     start_transport_stream_op_batch.
+     If this is used in preference to start_transport_stream_op_batch, the
+     following can be omitted also:
+       - calling init_call_elem, destroy_call_elem, set_pollset_or_pollset_set
+       - allocation of memory for call data
+     There is an on-going migration to move all filters to providing this, and
+     then to drop start_transport_stream_op_batch. */
+  grpc_core::ArenaPromise<grpc_core::TrailingMetadata> (*make_call_promise)(
```

Comment:
In the short-term world where some filters support this and others don't, how is this actually going to be used?  Will there be code in `grpc_call_next_op()` that checks whether the next filter is promise-aware and if so invokes `make_call_promise()` instead of `start_transport_stream_op_batch()`?  Or is the idea here that the channel stack will use this only if *all* filters in the stack support promises?

Reply:
The latter... we'll use it to decide that we use the promise based call code.
There's opportunity if we need it to also detect a run of promise based stuff at channel stack build time and wrap that run in a dynamically created filter - but it's unclear to me that we will or not.